flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/factories/HiveFunctionDefinitionFactory.java

```diff
@@ -67,14 +69,26 @@ public HiveFunctionDefinitionFactory(HiveConf hiveConf) {
 		this.hiveShim = HiveShimLoader.loadHiveShim(hiveVersion);
 	}
 
+	public HiveFunctionDefinitionFactory(String hiveVersion) {
```

Comment:
How about following the coding guidelines: https://flink.apache.org/contributing/code-style-and-quality-common.html#design-for-testability ? Especially the part: "Inject dependencies"

Reply:
what do you mean? can you elaborate?

Reply:
Yes, that's what I meant.